### PR TITLE
Enable passing in 'parallel' parameter to dotnet test

### PR DESCRIPTION
**Pull request by kichalla**
@natemcmaster @victorhurdugaci @pranavkm

### CODE REVIEW

File: KoreBuild-dotnet/build/_dotnet-test.shade [hunk 1]
```diff
@@ -18,7 +18,16 @@ default configuration = 'Debug'
 
 @{
     var projectFolder = Path.GetDirectoryName(projectFile);
```
Comment (kichalla) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: Following similar approach as in dnx test shade file

Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: Does it support parallel?

Comment (natemcmaster) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: Yes. Somehow the rumor started that dotnet test didn't, but this was just a bug. See https://github.com/dotnet/coreclr.xunit/pull/7

**Comment by natemcmaster** [comment]
:shipit:

File: KoreBuild-dotnet/build/_dotnet-test.shade (hunk 1 above)
Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: interesting. but if you run `dotnet test --help` you don't get the runner arguments

File: KoreBuild-dotnet/build/_dotnet-test.shade [hunk 2]
```diff
@@ -18,7 +18,16 @@ default configuration = 'Debug'
...
-    var testArgs = "--configuration " + configuration;
-    Dotnet("test " + testArgs, projectFolder);
+    var projectName = Path.GetFileName(projectFolder);
+
+    var noParallelTestProjects = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+    if (!string.IsNullOrEmpty(NO_PARALLEL_TEST_PROJECTS))
```
Comment (dougbu) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: Will probably need to restore the `NO_PARALLEL_TEST_PROJECTS` setting in at least the MVC repo.

File: KoreBuild-dotnet/build/_dotnet-test.shade (hunk 1 above)
Comment (kichalla) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: My guess is that `dotnet test` is a general purpose kick starter for a test runner, so you could have something like NUnit which might not support parallelism in which case this argument does not hold true.

Comment (natemcmaster) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: I believe dotnet test flows any unrecognized arguments to the testrunner. e.g. `--configuration Debug` is handled by `dotnet-test.exe` but `-parallel none` is passed to `dotnet-test-xunit.exe`. Order matters though: cref https://github.com/dotnet/cli/issues/1456

Comment (kichalla) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: > but if you run dotnet test --help you don't get the runner arguments

@victorhurdugaci Can you file a bug for this in dotnet cli?

Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: https://github.com/dotnet/cli/issues/1549

**Comment by citizenmatt** [comment]
Does this mean that test runners are not parallel by default? I.e. if I just run `dotnet test` for an xunit test project it will run single threaded?

**Comment by natemcmaster** [comment]
To see what is default, run `dotnet test -diagnostics`. It will show you the parallel setting it is using for that run.  
```
$ dotnet test -diagnostics
xUnit.net DNX Runner (64-bit osx.10.11-x64)
  Discovering: Microsoft.Data.Sqlite.Tests (method display = ClassAndMethod)
  Discovered:  Microsoft.Data.Sqlite.Tests (running 225 test cases)
  Starting:    Microsoft.Data.Sqlite.Tests (parallel test collections = on, max threads = 4)
```

**Comment by victorhurdugaci** [comment]
@citizenmatt that depends on the runner implementation. We don't control it.

**Pull request merged**